packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -474,20 +494,19 @@ export class EngineReportingAgent<TContext = any> {
     // `await` the `signature` if it's a Promise, prior to putting it
     // on the stack of traces to deliver to the cloud.
     const cachedSignature = await this.signatureCache.get(cacheKey);
+    let graphqlValidationFailure = false;
 
     if (cachedSignature) {
-      return cachedSignature;
+      return { signature: cachedSignature, graphqlValidationFailure};
     }
 
     if (!documentAST) {
```

Comment:
This condition is false if parsing fails, but not if validation fails. So it looks like you have consistently used the word "validation" to mean "parsing".

Reply:
Changed conditional logic.